gravitee-apim-integration-tests/src/test/java/io/gravitee/apim/integration/tests/messages/HttpPostEntrypointMqtt5EndpointIntegrationTest.java

```
@@ -0,0 +1,394 @@
+    }
+
+    @Override
+    public void configureEntrypoints(Map<String, EntrypointConnectorPlugin<?, ?>> entrypoints) {
+        entrypoints.putIfAbsent("http-post", EntrypointBuilder.build("http-post", HttpPostEntrypointConnectorFactory.class));
+    }
+
+    @Override
+    public void configurePolicies(Map<String, PolicyPlugin> policies) {
+        policies.put(
+            "interrupt-message-request-phase",
+            PolicyBuilder.build("interrupt-message-request-phase", InterruptMessageRequestPhasePolicy.class)
+        );
+        policies.put(
+               "assign-attributes",
+               PolicyBuilder.build("assign-attributes", AssignAttributesPolicy.class, AssignAttributesPolicyConfiguration.class)
+        );
+    }
+
+    @Override
+    public void configureApi(ReactableApi<?> api, Class<?> definitionClass) {
+        if (definitionClass.isAssignableFrom(Api.class)) {
+            Api apiDefinition = (Api) api.getDefinition();
+            apiDefinition
+                .getEndpointGroups()
+                .stream()
+                .flatMap(eg -> eg.getEndpoints().stream())
+                .filter(endpoint -> endpoint.getType().equals("mqtt5"))
+                .forEach(endpoint ->
+                    endpoint.setConfiguration(endpoint.getConfiguration().replace("mqtt5-port", Integer.toString(mqtt5.getMqttPort())))
+                );
+        }
+    }
+
+    @Test
+    void should_be_able_to_publish_to_mqtt5_endpoint_with_httppost_entrypoint(HttpClient client, VertxTestContext testContext) {
+        JsonObject requestBody = new JsonObject();
+        requestBody.put("field", "value");
+
+        final Mqtt5RxClient mqtt5RxClient = prepareMqtt5Client();
+
+        connectToMqtt5(testContext, mqtt5RxClient);
+
+        final TestSubscriber<Mqtt5Publish> mqttTestSubscriber = subscribeToMqtt5(mqtt5RxClient, TEST_TOPIC);
+
+        // Post a message
+        postMessage(client, "/test", requestBody, Map.of("X-Test-Header", "header-value"));
```

In order to avoid flaky test, I would have linked mqtt subscription and publication with for example something like :

connectToMqtt5()
.andthen(publish)
.test()
....

WDYT?

---

Indeed, I hesitated, here in the connectToMqtt5() method, I use a test observer and am doing the assertions on it with an await method.
I chose this way because (in my opinion) it was simpler to have simple steps that can fail fast and can be quickly identified compared to a big flow of operations.

But I would be happy to challenge this if you think it would be better

---

I replaced the assertions on connection and disconnection by blockingAwait and we will do some pair to rework that part properly